drivers/tee/optee/core.c

```diff
@@ -909,13 +910,19 @@ static void optee_smccc_smc(unsigned long a0, unsigned long a1,
 	g_smc_used_ring->tail = (g_smc_used_ring->tail + 1) % OPTEE_SHM_QUEUE_SIZE;
 	spin_unlock(&smc_ring_lock);
 
-	if (hypervisor_is_type(X86_HYPER_QNX)) {
-		*g_smc_evt_src = EVENT_KERNEL;
-		g_ivshmem_dev.ctrl->notify = 1 << g_smc_vm_ids->tee_id;
-	} else
-		writel(g_smc_vm_ids->tee_id << 16, g_ivshmem_dev.regs_addr + DOORBELL_OFF);
+	while (g_smc_args[index].a8 != OPTEE_HANDLE_DONE) {
+		if (++count > 60 )
```

Comment:
1. 60*2s = 2 min, too long. suggest 10 (for 20s) or 15(for 30s)
2. use "==" instead of ">" to avoid to many messages